x-pack/elastic-agent/pkg/agent/operation/operator.go

```diff
@@ -305,17 +305,19 @@ func (o *Operator) getApp(p Descriptor) (Application, error) {
 	var err error
 
 	monitor := o.monitor
+	appName := p.BinaryName()
 	if app.IsSidecar(p) {
 		// make watchers unmonitorable
 		monitor = noop.NewMonitor()
+		appName += "_monitoring"
 	}
 
 	if p.ServicePort() == 0 {
 		// Applications without service ports defined are ran as through the process application type.
 		a, err = process.NewApplication(
 			o.bgContext,
 			p.ID(),
-			p.BinaryName(),
+			appName,
```

Comment:
I wonder if this has any effect in other places except the status output? Hopefully a positive impact but we should double check in case we do any comparison on appName.

Reply:
From what I can see it should be safe to change it like this, the `Application` that is returned exposes it for use by other processes with the `Name` function